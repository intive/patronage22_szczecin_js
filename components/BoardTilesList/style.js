import styled from 'styled-components'

import { breakpoint } from '../../config/breakpoints'

export const List = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 2.5rem;
  & > * {
    margin-bottom: 2.5rem;
  }

  @media only screen and ${breakpoint.md} {
    flex-direction: row;
    flex-wrap: wrap;
    width: 36rem;
    margin: 2.5rem auto;
    & > * {
      margin: 0 1rem 2.5rem;
    }
  }

  @media only screen and ${breakpoint.lg} {
    width: 54rem;
    margin-top: 5rem;
  }

  @media only screen and ${breakpoint.xl} {
    width: 72rem;
    margin-top: 6rem;
  }
`
