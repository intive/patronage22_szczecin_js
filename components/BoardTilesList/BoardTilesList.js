import { useContext } from 'react'
import BoardTile from '../BoardTile/BoardTile'
import BoardsContext from '../../store/boards-context'
import CreateBoardTile from '../CreateBoardTile/CreateBoardTile'
import { List } from './style'

export default function BoardTilesList () {
  const boardsCtx = useContext(BoardsContext)
  const boards = boardsCtx.boards

  const generateBoardList = boards.map(board =>
    <BoardTile
      id={board.id}
      key={board.id}
      name={board.name}
      date={board.createdAt}
      cardCount={board.cardCount}
      hasPassword={board.hasPassword}
    />)

  return (
    <List data-testid='board-tiles-list'>
      <CreateBoardTile />
      {generateBoardList}
    </List>
  )
}
