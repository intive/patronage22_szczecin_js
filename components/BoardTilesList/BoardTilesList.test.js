
/* globals describe, it, expect  */

import { render } from '@testing-library/react'
import BoardTilesList from './BoardTilesList'

describe('BoardTilesList', () => {
  const tilesList = [
    {
      id: 'mockedId1',
      name: 'mockedName',
      date: '1 Jan 2022',
      cardCount: 0,
      hasPassword: false
    },
    {
      id: 'mockedId2',
      name: 'mockedName',
      date: '1 Jan 2022',
      cardCount: 3,
      hasPassword: true
    }
  ]

  it('should render name, date, no icon when password is not provided or render render icon when password is provided', () => {
    const { container } = render(<BoardTilesList tilesList={tilesList} />)
    expect(container).toMatchSnapshot()
  })
})
