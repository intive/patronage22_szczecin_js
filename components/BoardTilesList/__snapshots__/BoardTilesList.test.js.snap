// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardTilesList should render name, date, no icon when password is not provided or render render icon when password is provided 1`] = `
<div>
  <div
    class="sc-kfPuZi ebNvPM"
  >
    <div
      class="sc-bqiRlB bKrmmx"
      role="button"
      tabindex="0"
    >
      <button
        class="sc-crHmcD lefaQv"
      >
        <span
          class="material-icons sc-egiyK jjEWZz"
        >
          crop_original
        </span>
        New board
      </button>
    </div>
  </div>
</div>
`;
